coral-spark/src/main/java/com/linkedin/coral/spark/transformers/FuzzyUnionGenericProjectTransformer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.transformers;
+
+import java.net.URI;
+import java.util.Set;
+
+import org.apache.calcite.sql.SqlCall;
+
+import com.linkedin.coral.com.google.common.collect.ImmutableList;
+import com.linkedin.coral.common.functions.GenericProjectFunction;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.spark.containers.SparkUDFInfo;
+
+
+/**
+ * This transformer transforms the SqlCall `generic_project(col, col_name_string, hive_type_string)`
+ * (check {@link com.linkedin.coral.common.FuzzyUnionSqlRewriter}) to `generic_project(col, hive_type_string)`
+ * to meet Spark's expectation, and registers the `GenericProject` UDF.
+ */
+public class FuzzyUnionGenericProjectTransformer extends SqlCallTransformer {
+
+  private final Set<SparkUDFInfo> sparkUDFInfos;
```

Comment:
I think `final` is fine, as it does not mean the set is immutable, only make `sparkUDFInfos` cannot be assigned to another set.

Reply:
`final` doesn't mean this is immutable, it means the reference to it can't be changed.
https://stackoverflow.com/questions/26500423/what-does-it-mean-for-a-collection-to-be-final-in-java